src/field_arithmetic.rs

```diff
@@ -35,10 +35,10 @@ impl<F: Field> GadgetBuilder<F> {
     }
 
     /// `x^p` for a constant `p`.
-    pub fn exp(&mut self, x: &Expression<F>, p: &Element<F>) -> Expression<F> {
+    pub fn exponentiation(&mut self, x: &Expression<F>, p: &Element<F>) -> Expression<F> {
         // This is exponentiation by squaring. For each 1 bit of p, multiply by the associated
         // square power.
-        let mut product = Expression::one();
+        let mut product_exp = Expression::one();
```

Comment:
Appended with `_exp` for notation consistency.

Reply:
This is where a note in a `readme` works best. :) I'm OK with any standard that we go with, and I'll make sure to reformat the code to it.